test/java/seedu/addressbook/commands/ViewAllCommandTest.java

```diff
@@ -0,0 +1,150 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.util.TestUtil;
+
+public class ViewAllCommandTest {
+    private AddressBook addressBook;
+    private AddressBook emptyAddressBook;
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> listWithAll;
+    private List<ReadOnlyPerson> listWithSome;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"),
+                                    new Phone("98765432", false),
+                                    new Email("johnd@gmail.com", false),
+                                    new Address("John street, block 123, #01-01", false),
+                                    new UniqueTagList(Collections.emptySet()));
+        Person betsyCrowe = new Person(new Name("Betsy Crowe"),
+                                       new Phone("1234567", true),
+                                       new Email("betsycrowe@gmail.com", false),
+                                       new Address("Newgate Prison", true),
+                                       new UniqueTagList(new Tag("friend"), new Tag("criminal")));
```

Comment:
Two persons is a bit a special case because the list doesn't have persons in the middle. A 3-person list is more general.